src/components/Image/OSD/OSD.tsx

```diff
@@ -0,0 +1,100 @@
+import {
+  Navigator,
+  Viewport,
+  Wrapper,
+} from "src/components/Image/Image.styled";
+import OpenSeadragon, { Options } from "openseadragon";
+import React, { useEffect, useState } from "react";
+
+import Controls from "src/components/Image/Controls/Controls";
+import { OpenSeadragonImageTypes } from "src/types/open-seadragon";
+import { getInfoResponse } from "src/lib/iiif";
+
+interface OSDProps {
+  _cloverViewerHasPlaceholder: boolean;
+  ariaLabel?: string | null;
+  config: Options;
+  uri: string | undefined;
+  imageType: OpenSeadragonImageTypes;
+  openSeadragonCallback?: (viewer: OpenSeadragon.Viewer) => void;
+}
+
+const OSD: React.FC<OSDProps> = ({
+  ariaLabel,
+  config,
+  uri,
+  _cloverViewerHasPlaceholder,
+  imageType,
+  openSeadragonCallback,
+}) => {
+  const [osdUri, setOsdUri] = useState<string>();
+  const [openSeadragon, setOpenSeadragon] = useState<OpenSeadragon.Viewer>();
+
+  useEffect(() => {
+    if (!openSeadragon) setOpenSeadragon(OpenSeadragon(config));
+
+    return () => openSeadragon?.destroy();
+  }, []);
+
+  useEffect(() => {
+    if (openSeadragon && openSeadragonCallback)
+      openSeadragonCallback(openSeadragon);
+  }, [openSeadragon, openSeadragonCallback]);
+
+  useEffect(() => {
+    if (uri !== osdUri) setOsdUri(uri);
+  }, [osdUri, uri]);
+
+  useEffect(() => {
+    if (osdUri && openSeadragon) {
+      switch (imageType) {
+        case "simpleImage":
+          openSeadragon.addSimpleImage({
+            url: osdUri,
+          });
+          break;
+        case "tiledImage":
+          getInfoResponse(osdUri).then((tileSource) => {
+            openSeadragon.addTiledImage({
+              tileSource: tileSource,
+            });
+          });
+          break;
+        default:
+          openSeadragon.close();
+          console.warn(
+            `Unable to render ${osdUri} in OpenSeadragon as type: "${imageType}"`,
+          );
+          break;
+      }
+    }
+  }, [osdUri, openSeadragon]);
```

Comment:
Include `imageType` in the dependency array?

Reply:
👍